tsl/src/fdw/deparse.c

```diff
@@ -867,6 +867,60 @@ deparseDistinctClause(StringInfo buf, deparse_expr_cxt *context)
 			return;
 	}
 
+	if (query->hasDistinctOn)
+	{
+		/*
+		 * Pushing down DISTINCT ON is more complex than plain DISTINCT.
+		 * The DISTINCT ON columns must be a prefix of the ORDER BY columns.
```

Comment:
```suggestion
		 * The DISTINCT ON columns must be the prefix of the ORDER BY columns.
```

Reply:
ORDER BY has many prefixes of different length, and DISTINCT ON has to be one of these prefixes. That's an "a". We know that it is "the" particular prefix that has matching length, but that's not specified anywhere in the context or the sentence itself. In the next sentence, we could write "the prefix", because now we would be talking about the particular one that matches DISTINCT ON.